jspwiki-main/src/main/java/org/apache/wiki/providers/VersioningFileProvider.java

```diff
@@ -685,11 +685,11 @@ private static class CachedProperties {
          */
         public CachedProperties( final String pageName, final Properties props, final long lastModified ) {
             if ( pageName == null ) {
-                throw new NullPointerException ( "pageName must not be null!" );
+                throw new IllegalArgumentException ( "pageName must not be null!" );
```

Comment:
I think this is one of those cases where `NullPointerException` really expresses the intent - the field must not be null